packages/opentelemetry-plugin-xml-http-request/src/xhr.ts

```diff
@@ -0,0 +1,371 @@
+      const callbackToRemoveEvents = spanId
+        ? plugin._callbackToRemoveEvents[spanId]
+        : undefined;
+
+      if (typeof callbackToRemoveEvents === 'function') {
+        callbackToRemoveEvents();
+      }
+      const currentSpan = spanId ? plugin._spans[spanId] : undefined;
+      if (currentSpan) {
+        plugin._addSpanNetworkEvents(currentSpan);
+        currentSpan.addEvent(eventName);
+
+        const url = currentSpan.attributes[AttributeNames.HTTP_URL];
+        if (typeof url === 'string') {
+          const parsedUrl = parseUrl(url);
+
+          currentSpan.setAttribute(AttributeNames.HTTP_STATUS_CODE, xhr.status);
+          currentSpan.setAttribute(
+            AttributeNames.HTTP_STATUS_TEXT,
+            xhr.statusText
+          );
+          currentSpan.setAttribute(
+            AttributeNames.HTTP_HOSTNAME,
+            parsedUrl.hostname
+          );
+          currentSpan.setAttribute(
+            AttributeNames.HTTP_PATH,
+            parsedUrl.pathname
+          );
+
+          // @TODO do we want to collect this or it will be collected earlier once only or
+          //    maybe when parent span is not available ?
+          currentSpan.setAttribute(
+            AttributeNames.HTTP_USER_AGENT,
+            navigator.userAgent
+          );
+        }
+
+        currentSpan.end();
+      }
+    }
+
+    function onError(this: XMLHttpRequestWrapped) {
+      endSpan(this, EventNames.EVENT_ERROR);
+    }
+
+    function onTimeout(this: XMLHttpRequestWrapped) {
+      endSpan(this, EventNames.EVENT_TIMEOUT);
+    }
+
+    function onLoad(this: XMLHttpRequestWrapped) {
+      if (this.status < 299) {
+        endSpan(this, EventNames.EVENT_LOAD);
+      } else {
+        endSpan(this, EventNames.EVENT_ERROR);
+      }
+    }
+
+    function unregister(xhr: XMLHttpRequestWrapped, spanId: string) {
+      xhr.removeEventListener('load', onLoad);
+      xhr.removeEventListener('error', onError);
+      xhr.removeEventListener('timeout', onTimeout);
+      delete plugin._callbackToRemoveEvents[spanId];
+      delete xhr.__OT_SPAN_ID;
+    }
+
+    return (original: SendFunction): SendFunction => {
+      return function patchSend(
+        this: XMLHttpRequest,
+        body?: SendBody
+      ): SendFunction {
+        const spanId = (this as XMLHttpRequestWrapped).__OT_SPAN_ID;
+        const currentSpan: types.Span | undefined = spanId
+          ? plugin._spans[spanId]
+          : undefined;
+
+        if (currentSpan && spanId) {
+          currentSpan.addEvent(EventNames.METHOD_SEND);
+
+          this.addEventListener('load', onLoad);
+          this.addEventListener('error', onError);
+          this.addEventListener('timeout', onTimeout);
+
+          plugin._callbackToRemoveEvents[spanId] = () => {
+            unregister(this as XMLHttpRequestWrapped, spanId);
+          };
+          plugin._addHeaders(this, currentSpan);
+        }
+        return original.call(this, body);
```

Comment:
```suggestion
        return original.apply(this, arguments);
```

Reply:
1. call is faster
2. we already know there is just one para,  we know it's type so we can use it and typescript will also validate the type of this is correcty
3. You don't have `arguments` word in typescript , so you would still declare array of aguments and define type for it, which would be quite weird anyway

Reply:
and the same here, thx
 @draffensperger @mayurkale22 ^^